sdk/schemaregistry/azure-schemaregistry-avroserializer/azure/schemaregistry/serializer/avroserializer/_avro_serializer.py

```diff
@@ -38,8 +42,16 @@ def __init__(self, codec=None):
         :param str codec: The writer codec. If None, let the avro library decides.
         """
         self._writer_codec = codec
-        self._schema_writer_cache = {}  # type: Dict[str, DatumWriter]
-        self._schema_reader_cache = {}  # type: Dict[str, DatumReader]
+
+    @lru_cache(maxsize=128)
+    def _get_schema_writer(self, schema):
+        schema = avro.schema.parse(schema)
```

Comment:
Because the lru_cache only takes hashable types, we have to pass in the str(schema) into this and parse again. I'm not sure whether this is the best option or whether we could use a dict cache like before, but just add checks for removing if >128 items?

Reply:
I think it should be fine, if we hit the cache, then there's no parsing.
it's a one-time cost to put into cache